art/defences/preprocessor/resample.py

```diff
@@ -0,0 +1,133 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the resampling defence `Resample`.
+
+| Please keep in mind the limitations of defences. For details on how to evaluate classifier security in general,
+    see https://arxiv.org/abs/1902.06705  .
+"""
+
+import logging
+
+import numpy as np
+
+from art.defences.preprocessor.preprocessor import Preprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class Resample(Preprocessor):
+    """
+    Implement the resampling defense approach.
+
+    Resampling implicitly consists of a step that applies a lowpass filter. The underlying filter in
+    this implementation is a windowed sinc interpolation function.
+    """
+
+    params = ["sr_original", "sr_new", "channel_index"]
+
+    def __init__(
+        self, sr_original, sr_new, channel_index=2, apply_fit=True, apply_predict=False
+    ):
+        """Create an instance of MP3 compression.
+
+        :param sr_original: Original sampling rate of sample.
+        :type sr_original: `int`
+        :param sr_new: New sampling rate of sample.
+        :type sr_new: `int`
+        :param channel_index: Index of the axis containing the audio channels.
+        :type channel_index: `int`
+        :param apply_fit: True if applied during fitting/training.
+        :type apply_fit: `bool`
+        :param apply_predict: True if applied during predicting.
+        :type apply_predict: `bool`
+        """
+        super(Resample, self).__init__()
```

Comment:
```suggestion
        super().__init__()
```

Reply:
Right, this is equivalent in Python 3 and your suggestion is lesser lines :+1: